lib/middleware.js

```diff
@@ -153,6 +161,7 @@ module.exports = function(self, options) {
       if (candidates.length === 1) {
         setLocale(candidates[0]);
       }
+      // todo .. locale isnt here anymore
```

Comment:
@boutell how should this get handled?

Reply:
It's in a cookie now, so get it from the cookie. If it's there (it might not be because of the option to not use the cookie); otherwise leave req.locale untouched.

Reply:
We were going in a direction that didn't make all that much sense because it would not work if domains were configured (cookies are not cross-domain), but not yet fully configured to distinguish all locales (so we can't guess from the URL). And the existing req.session.locale setup DOES work in that situation, because of our cross-domain session tokens. And... yeah.

A simplification is in order:

If csrf.disableAnonSession is true, we don't touch req.session unless the user is logged in. Otherwise we do it exactly the way we do it now.

Along with this, we document that if you use disableAnonSession, you MUST completely distinguish your locales by hostname and/or prefix, which is really indispensable for SEO anyway, or else the general public won't be able to switch locales reliably.

THE END

So I'm going to close this PR, with appreciation, and pursue this simplified approach I wish I'd recognized the need for in the beginning.